### PR TITLE
Combine the GitHub workflows for pull requests.

### PR DESCRIPTION
**Pull request by motlin**
Before:

Screenshot 2024-05-19 at 10 58 04 PM

After:

Screenshot 2024-05-19 at 10 58 21 PM

Before:

Screenshot 2024-05-19 at 10 59 31 PM

After:

Screenshot 2024-05-19 at 10 59 45 PM

### CODE REVIEW

**Comment by motlin** [comment]
It's best to organize GitHub workflows so that there's one per trigger (for us that will be pull request and cron) and then have several jobs within each workflow.

These changes are stacked on #1585 and must land afterwards.

File: .github/workflows/pull-request.yml [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    steps:
+      - uses: actions/checkout@v4
+      - run: mvn --non-recursive wrapper:wrapper -Dmaven=3.9.6
+      - name: Set up Java 11
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'zulu'
+          java-version: 11
+      - name: JavaDoc Aggregate
+        run: ./mvnw -Dsurefire.useFile=false -DskipTests -Denforcer.skip=true clean install javadoc:aggregate --projects '!unit-tests,!serialization-tests,!jcstress-tests,!junit-trait-runner,!unit-tests-java8,!test-coverage,!p2-repository,!org.eclipse.collections:org.eclipse.collections'
```
Comment (donraab) [inline .github/workflows/pull-request.yml:126]: @motlin , you changed from `-pl` to `--projects` for both run commands here.

File: .github/workflows/pull-request.yml [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+      - name: 'Print outputs'
+        shell: bash
+        run: |
+          echo 'Outputs'
+          echo "steps.download-jdk.outputs.archive = ${{ steps.download-jdk.outputs.archive }}"
+          echo "steps.download-jdk.outputs.version = ${{ steps.download-jdk.outputs.version }}"
+      - run: mvn --non-recursive wrapper:wrapper -Dmaven=3.9.6
+      - run: java --version
+      - name: Javadoc
+        run: ./mvnw -Dsurefire.useFile=false -DskipTests -Denforcer.skip=true clean install javadoc:aggregate -pl '!unit-tests,!serialization-tests,!jcstress-tests,!junit-trait-runner,!unit-tests-java8,!test-coverage,!p2-repository,!org.eclipse.collections:org.eclipse.collections'
```
Comment (donraab) [inline .github/workflows/pull-request.yml:154]: @motlin , you left this one as `-pl `instead of `--projects` as you have above in the javadoc job section.

**Review by donraab — CHANGES_REQUESTED** [review]
Minor inconsistency between javadoc and javadocearlyaccess

File: .github/workflows/pull-request.yml (hunk 2 above)
Comment (motlin) [reply .github/workflows/pull-request.yml]: @donraab replaced both `-pl` with `--projects`

**Pull request merged**